source/common/event/file_event_impl.cc

```diff
@@ -39,7 +40,14 @@ void FileEventImpl::activate(uint32_t events) {
   event_active(&raw_event_, libevent_events, 0);
 }
 
+void FileEventImpl::assignEventBase(event_base* base) {
+  // we cannot use below
+  // ASSERT(-1 !=event_base_set(base, &raw_event_));
+  raw_event_.ev_base = base;
+}
+
 void FileEventImpl::assignEvents(uint32_t events) {
+  ASSERT(base_ == event_get_base(&raw_event_));
```

Comment:
@mattklein123 I want to demonstrate that `FileEventImpl::base_` is not required. 
The reason is that `base_` could be extract from `raw_event_`.
So we can remove the member `base_` from `FileEventImpl` and get 8 bytes back.

Reply:
Yes, OK, this makes sense, though I would implement it slightly differently vs. directly accessing the ev_base member. Basically just modify assignEvents() to take a base as a parameter. The first time it's called, pass the base from the constructor, then in `setEnabled()` pass the base from the event itself.

Please update the PR title, description, etc. to be more clear. Thanks.

/wait

Reply:
SG. Will follow your suggestion and update. 
Thanks!